src/basic/macro.h

```diff
@@ -296,6 +296,9 @@ static inline int __coverity_check_and_return__(int condition) {
              p != (typeof(p)) POINTER_MAX;                                               \
              p = *(++_l))
 
+#if HAVE_THREADS_H
+#include <threads.h>
```

Comment:
i don't think we should include that here. Let the consumers do that.

The fact that threads.h exists is enough really, we don't need to also import it here.

(This is super generic and basic macro.h after all, which we should be very conservative with importing stuff from)

(and yes, this might mean we need to add includes of threads.h to many users of thread_local, but I think that's how god intended it)

Reply:
Did that. (include threads.h on callers)  but that would raise glibc dependency to 2.18 or later. is that ok ?

Reply:
why would it do that? as long as the users of `thread_local` add both:

```c
#if HAVE_THREADS_H
#include <threads.h>
#endif

#include "macro.h"
```

we should be good and not require any new deps. It's arguably what they should always have done...

Reply:
Yes, but then you no longer have a thread_local definition for the older (obsolete?)  systems.. so the meson.build check must define thread_local if the header is not found or similar..
Maybe I am not been clear enough.. or mising soemthing.. . 
So. let's use your suggestion

```c
#if HAVE_THREADS_H
#include <threads.h>
#endif

#include "macro.h"
```
Let see what macro.h does

```c
#ifndef thread_local
 
/* TRUE when -std=gnu2x.. compiler **does not define it as macro** neither the library does if  __STDC_VERSION__ > 201710L  && GCC >= 13 !!  */
 
#if __STDC_VERSION__ >= 201112L  &&

/* __STDC_VERSION == 202000L  condition true...*/

!(defined(__STDC_NO_THREADS__) || (defined(__GNU_LIBRARY__) && __GLIBC__ == 2 && __GLIBC_MINOR__ < 16))
 /* !(false || ( true && true && false))  --> !(false || false ) ---> !(false)  --> true  */

/* so #if true && true ..thread_local is redefined ..*/
#define thread_local _Thread_local
#else
#define thread_local __thread
#endif
#endif
```
and we are back to the issue unless some modification is made to macro.h, with clang is easy as it has __is_identifier to differentiate keywords from identifiers but gcc does not. need to change macro.h preprocessor hackery and that's exactly what was I avoiding :-)

Reply:
hmm, my thinking was to change macro.h to this:

```c
/* If threads.h doesn't exist, then define our own thread_local to match C11's thread_local. If it exists we expect potential users to include that header file directly */
#if !HAVE_THREADS_H
#ifndef thread_local
/*
 * Don't break on glibc < 2.16 that doesn't define __STDC_NO_THREADS__
 * see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=53769
 */
#if __STDC_VERSION__ >= 201112L && !(defined(__STDC_NO_THREADS__) || (defined(__GNU_LIBRARY__) && __GLIBC__ == 2 && __GLIBC_MINOR__ < 16))
#define thread_local _Thread_local
#else
#define thread_local __thread
#endif
#endif
```

Reply:
btw, might make sense to move this definition to src/basic/missing_threads.h matching our other src/basic/missing_xyz.h files

Reply:
> btw, might make sense to move this definition to src/basic/missing_threads.h matching our other src/basic/missing_xyz.h files

Yes I Thought about doing that, it just does not belong in the macro.h header.